integration-tests/src/test/java/io/testerra/test/pretest_status/DataProviderTest.java

```diff
@@ -30,20 +29,30 @@
 public class DataProviderTest extends TesterraTest {
 
     @Test(dataProviderClass = MyDataProvider.class, dataProvider = "throwException")
-    @Fails
-    public void test_interceptCrashedDataProvider() {
+    public void testT01_interceptCrashedDataProvider() {
         Assert.assertTrue(true);
     }
 
     @DataProvider()
     public Object[][] crashingDataProvider() {
-        throw new RuntimeException("crashed data provider");
+        throw new RuntimeException("Crashed data provider");
     }
 
     @Test(dataProvider = "crashingDataProvider")
-    @Fails
-    public void test_crashedDataProvider(Object object) {
+    public void testT02_crashedDataProvider(Object object) {
         Assert.assertTrue(true);
     }
 
+    @DataProvider(name = "assertFailedDataProvider")
+    public Object[][] dpAssertFailedMethod() {
+        Object[][] objects = new Object[1][1];
+        Assert.fail("Failed assertion in data provider");
```

Comment:
How is this different from `crashingDataProvider`? ... _Is_ it, incidentally? _Should_ it be?

Reply:
We have a dataproider with an exception and we have a dataprovider with an assertion.

Reply:
Ah, `org.testng.Assert` throws `java.lang.AssertionError`! Okay, fair enough, that's enough of a difference for me.
... I am accustomed to test assertions throwing either `IllegalArgumentException` or `IllegalStateException` - both of which are `RuntimeException`s.